src/webview/html/messageHeaderAsHtml.js

```diff
@@ -62,7 +62,7 @@ export default (
 
   if (item.type === 'stream' && headerStyle === 'full') {
     const streamName = streamNameOfStreamMessage(item);
-    const stream = subscriptions.find(x => x.name === streamName);
+    const stream = [...subscriptions.values()].find(x => x.name === streamName);
```

Comment:
The change at this line risks being significantly slower: the old code is linear (which isn't good), and the new code is still linear, but the new code does a linear amount of *allocation*, building up this new array, and that's potentially quite a bit more costly than just searching through an array.

And this line is in the code that generates the HTML for the message list, so it's something we potentially do many times if the user is looking at a view with a lot of history (provided it's an interleaved view where the stream name appears in headers -- for example the @-mentions or starred-messages views, or a search view once we have #4859.)

Happily, I think there is an easy solution! We already have the stream ID here, because `item` is a `Message | Outbox` with `type: 'stream'`, and those have a `stream_id`.

So we can just use `subscriptions.get(item.stream_id)`, much like in the spot in `action-sheets/index.js` that motivates this change.

Reply:
Hmm, I guess `Outbox` doesn't actually have `stream_id` yet: #3998. (Since #4667 it's there on the type as an optional property, but we never actually provide it.) So that simple solution won't yet work.

The most direct solution here would be to use the `subscriptions.values()` [iterator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators) just like you're doing, but use it directly with something like a for/of loop, to avoid allocating a big array of all the elements.

Another solution would be to go ahead and pass down a `subscriptionsByName` value into this code, gotten from `getSubscriptionsByName`.

Reply:
using `for ... of`.